### PR TITLE
feat(test): support travis

### PR DESCRIPTION
**Pull request by maxhbr**
This contains a simple configuration to use travis to build sw360-dependencies and the corresponding containers.

It tests whether at the end the nginx reponds

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  - set -e
+  - cd deployment
+  - echo "$CONFIGURATION_ENV" >> configuration.env
+  - ./docker-compose.sh --prepare
+  - ./docker-compose.sh build
+  - ./docker-compose.sh up -d
+  - sleep 120
+  - ./docker-compose.sh logs sw360 | grep -iq "Determine dialect for PostgreSQL 9"
+  - ./docker-compose.sh logs sw360couchdb | grep -iq "Apache CouchDB has started"
+  - curl -L -S -k https://localhost:8443 | grep -iq "html"
```
Comment (lepokle) [inline .travis.yml]: This test is not sufficient. e.g. the bad gateway page from ngnix also includes "html".

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+# Part of the SW360 Portal Project.
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+#
+# initial author: maximilian.huber@tngtech.com
+# -----------------------------------------------------------------------------
+# dist: precise
```
Comment (lepokle) [inline .travis.yml]: superfluous line?

File: .travis.yml (hunk 1 above)
Comment (lepokle) [inline .travis.yml]: Maybe one check for each container would be great? E.g. couch db can also be tested very easily.

**Review by lepokle — CHANGES_REQUESTED** [review]
Please check comments. At least the tomcat check must be better.

**Comment by maxhbr** [comment]
The tests currently fail randomly. The following two tests were executed on the same configuration with the same code:
- https://travis-ci.org/sw360/sw360chores/jobs/251542658
  - FAILED at 
    ```
    curl -S -k https://localhost:8443 | grep -iq "<title>Welcome - SW360</title>"
    ```
- https://travis-ci.org/sw360/sw360chores/jobs/251542652
  - SUCCESS

##

File: deployment/docker-compose.sh [hunk 3]
```diff
@@ -89,7 +89,7 @@ addSudoIfNeeded() {
     }
 }
 
-cmdDocker="$(addSudoIfNeeded) env $(grep -v '^#' proxy.env | xargs) docker"
+cmdDocker="$(addSudoIfNeeded) env $(grep -v '^#' "$DIR/proxy.env" | xargs) docker"
```
Comment (maxhbr) [inline deployment/docker-compose.sh:92]: This change allows the `docker-compose.sh` script to be called from arbitrary folders and not only as `./docker-compose.sh`

File: packaging/Rakefile [hunk 4]
```diff
@@ -21,7 +21,7 @@ task :default => :all
 def wget(urls)
   urls.each do |url|
     mkdir_p CACHE_DIR
-    sh "wget -c -N -P #{CACHE_DIR} #{url}"
+    sh "wget -q -c -N -P #{CACHE_DIR} #{url}"
```
Comment (maxhbr) [inline packaging/Rakefile:24]: Without the quiet flag, wget produces to many lines of output which surpass the travis limit.

**Comment by maxhbr** [comment]
**observation:** the tests which fail are chosen randomly (at least it seems so), but if a test has failed, it always fails and restarting does not help.

**Review by alexbrdn — COMMENTED** [review]
As far as I can tell, the code looks good. If only it would work reliably...

**Comment by maxhbr** [comment]
superfluous due to #48

**Pull request closed without merging**